exporter/debugexporter/README.md

```diff
@@ -107,6 +108,38 @@ Attributes:
         {"kind": "exporter", "data_type": "traces", "name": "debug"}
 ```
 
+## Using the collector's internal logger
+
+When `use_internal_logger` is set to `true` (the default), the exporter uses the collector's [internal logger][internal_telemetry] for output.
+This comes with several consequences:
+
+1. The exporter's output is prefixed and suffixed by the internal logger's data.
+   - The prefix is the current timestamp and the `info` level marker, similar to `2023-11-10T22:49:03.510-0600    info`
+   - The suffix is the exporter's metadata in JSON format, similar to `{"kind": "exporter", "data_type": "traces", "name": "debug"}`
+1. The exporter's output is sent to the same destination as the collector's logs. By default, it is the [standard error][stderr] stream,
+   configurable via the [service::telemetry::logs::output_paths][configure_internal_logs] option.
+   It is not possible to send the Debug exporter's output to a different stream than the collector's internal logs.
+1. The output from the Debug exporter is sent to the internal logger with the log level `INFO`.
+   This means that if user configures the `service::telemetry::logs::level` to `WARN` or `ERROR`,
+   they will not see the output from the Debug exporter.
+   As a result, it is not possible to configure the collector to see the Debug exporter's output
+   and mute collector's INFO logs at the same time.
+1. The exporter's output is subject to the internal logger's sampling,
+   which is separate from the sampling defined in the exporter's configuration.
+   See https://github.com/open-telemetry/opentelemetry-collector/issues/9921 and the note in the "Workaround" section of the issue.
```

Comment:
I am not super at ease with this section. Some of the things mentioned depend on the logger encoding. Some other things may change in the future, since there no guarantee that we will keep the format of logs in the non-machine readable format.

Can we instead just mention that this will use the Collector's internal logger and honor the `service::telemetry::logs` configuration? We can clarify how the Collector internal logger works in a separate place if needed, but I don't think this is the place to do so.

Reply:
Yeah that works, I'll change the description.

Reply:
Updated the documentation in https://github.com/open-telemetry/opentelemetry-collector/pull/10227/commits/02e356d7d6f2f791eff8435b91661cd002c5d3dc, please let me know your thoughts @mx-psi.